### ISSUE TITLE
Links in the preview are broken.

### ISSUE THREAD

**Issue by vorlif** [wagtail/wagtail#9123]
First of all, thanks to everyone for the work on version 4, it turned out great.

### Issue Summary

When I click on a link within the preview, I get an error message. To get rid of the error message, I have to reload the page. I did not expect to be able to navigate through the page, but there should not be an error message.

![grafik](https://user-images.githubusercontent.com/4333770/187940848-4c5dcbd1-912f-4252-a1cd-a78f1afd3306.png)

### Steps to Reproduce

1. Install [bakerydemo](https://github.com/wagtail/bakerydemo)
2. Start server (`./manage.py runserver`) and login as admin
3. Open the preview
4. Click a link

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: no

### Technical details

-   Python version: Python 3.10.6
-   Django version: 4.0.7
-   Wagtail version: 4.0
-   Browser version: 
* Firefox 104.0.1 (64-bit)
* Chrome 105.0.5195.52 (Official Build) (64-bit)

**Comment by laymonage** [comment]
Hey, thanks for the report!

I'm not sure if this is something we should handle on Wagtail's side, but the error is because the real page (not-preview) has the `X-Frame-Options` header set to `DENY`.

There are some workarounds to this:
- In your base template, add the following [`<base>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base) element within the `<head>` element, before any `<link>` elements:
  ```html+django
  {% if request.in_preview_panel %}
      <base target="_blank">
  {% endif %}
  ```
  This will make all links in the preview panel open in a new tab. You can also use `_parent` if you prefer the link to be opened in the same tab.
- Or, [set your `X-Frame-Options` to `SAMEORIGIN`](https://docs.djangoproject.com/en/4.1/ref/clickjacking/#setting-x-frame-options-for-all-responses). This does not handle external links, so the issue will still show up if the external site doesn't allow iframes.

I think it's also possible for Wagtail to modify all links in the iframe to use `target="_blank"` or disable all links within the iframe, but I don't think it's a good thing to do...

**Comment by vorlif** [comment]
Hi, thanks for your answer and analysis. I should have thought of that myself. :see_no_evil: 

I am unsure if you should be able to navigate within the preview or not. If you think this should not be handled by Wagtail, feel free to close the issue.

**Comment by gasman** [comment]
> * In your base template, add the following [`<base>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base) element within the `<head>` element, before any `<link>` elements:
>   ```
>   {% if request.in_preview_panel %}
>       <base target="_blank">
>   {% endif %}
>   ```

I think this would be a worthwhile addition to [base.html in the project template](https://github.com/wagtail/wagtail/blob/main/wagtail/project_template/project_name/templates/base.html).

**Comment by laymonage** [comment]
PR #9132, I've also added an upgrade consideration with the above code snippet.

**Comment by gasman** [comment]
Fixed in #9132

**Comment by krzysztofjeziorny** [comment]
Thanks @laymonage for pointing out the `x-frame-options` header.

The `DENY` setting is on in quite popular boilerplate projects for both [Nginx](https://github.com/h5bp/server-configs-nginx) and [Apache](https://github.com/h5bp/server-configs-apache).
